src/lib.rs

```diff
@@ -214,31 +204,57 @@ impl Opts {
   }
 }
 
+#[derive(clap::ValueEnum, Clone, Debug)]
+pub enum OptArgs {
```

Comment:
All this CLI interface code should be moved to `main.rs`.
hvml as a library shouldn't be aware of CLI

Reply:
Can't move it to `main.rs`, the only option I see is to crate a public `cli` crate

Reply:
You can move it, just need to not have a impl block for Opts there

Reply:
You could instead have a `OptArgs::modify_opts(&self, &mut Opts) -> ()` if you want to keep it a method, and then `cli_opts.for_each(|o| o.modify_opts(opts))`